inc/Engine/Admin/RocketInsights/Jobs/Manager.php

```diff
@@ -46,21 +47,39 @@ class Manager implements ManagerInterface, LoggerAwareInterface {
 	 */
 	protected $plan;
 
+	/**
+	 * API Client instance.
+	 *
+	 * @var APIClient
+	 */
+	protected $api_client;
+
+	/**
+	 * Timeout value in seconds for synchronization operations.
+	 *
+	 * This constant defines the maximum time allowed for sync jobs
+	 * before they are considered to have timed out.
+	 */
+	const SYNC_TIMEOUT = 10;
+
 	/**
 	 * Instantiate the class.
 	 *
 	 * @param RocketInsightsQuery $query Rocket Insights Query instance.
 	 * @param ContextInterface    $context Rocket Insights Context.
 	 * @param Plan                $plan Plan instance.
+	 * @param APIClient           $api_client API Client instance.
 	 */
 	public function __construct(
 		RocketInsightsQuery $query,
 		ContextInterface $context,
-		Plan $plan
+		Plan $plan,
+		APIClient $api_client
 	) {
-		$this->query   = $query;
-		$this->context = $context;
-		$this->plan    = $plan;
+		$this->query      = $query;
+		$this->context    = $context;
+		$this->plan       = $plan;
+		$this->api_client = $api_client;
```

Comment:
We would no longer need this as a dependency if we use the `send_api` from `JobProcessor`